src/v/kafka/server/materialized_partition.h

```diff
@@ -96,6 +96,37 @@ class materialized_partition final : public kafka::partition_proxy::impl {
           : error_code::offset_out_of_range;
     }
 
+    result<partition_info> get_partition_info() const final {
```

Comment:
is an implementation here necessary for something? i was under the impression that we didn't need to maintain the materialized partition integration?

Reply:
thanks to your PR from few days ago it is not, going to drop it

Reply:
i added a commit that removes the `materlized_partition` abstraction completely